src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -277,18 +277,23 @@ public static AzureVMDeploymentInfo createDeployment(final AzureVMAgentTemplate
             if (StringUtils.isNotBlank(storageAccountName)) {
                 ObjectNode.class.cast(tmp.get("variables")).put("storageAccountName", storageAccountName);
             }
-            
+
             if(StringUtils.isNotBlank(blobEndpointSuffix)){
                 ObjectNode.class.cast(tmp.get("variables")).put("blobEndpointSuffix", blobEndpointSuffix);
             }
-                
+
             // Network properties.  If the vnet name isn't blank then
             // then subnet name can't be either (based on verification rules)
             if (StringUtils.isNotBlank(template.getVirtualNetworkName())) {
                 ObjectNode.class.cast(tmp.get("variables")).put("virtualNetworkName", template.getVirtualNetworkName());
                 ObjectNode.class.cast(tmp.get("variables")).put("subnetName", template.getSubnetName());
+                if (StringUtils.isNotBlank(template.getVirtualNetworkResourceGroupName())) {
+                    ObjectNode.class.cast(tmp.get("variables")).put("virtualNetworkResourceGroupName", template.getVirtualNetworkResourceGroupName());
+                } else {
+                    ObjectNode.class.cast(tmp.get("variables")).put("virtualNetworkResourceGroupName", resourceGroupName);
+                }
             } else {
-                AddDefaultVNetResourceNode(tmp, mapper);
+                AddDefaultVNetResourceNode(tmp, mapper, resourceGroupName);
```

Comment:
In this branch the vNet is blank and we use the default `resourceGroupName`.

So I think it's correct?